test/host_tests/sal_tcpclient.py

```diff
@@ -0,0 +1,110 @@
+"""
+mbed OS
+Copyright (c) 2011-2016 ARM Limited
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+
+import sys
+import socket
+import time
+import threading
+import re
+from mbed_host_tests import BaseHostTest
+
+
+class SalTcpClientTest(BaseHostTest):
+    """
+    mbed greentea framework host test script for tcp_echo_server client 
+    side functionality. The test does the following:
+      - Receives over the serial port the target ipaddr, port of the 
+        target tcp server
+      - establishes a connection with the remoted tcp echo server
+      - sends ever increasing data buffers using send() starting with
+        a 16 byte buffer and doubling the size of the data sent on each 
+        iteration through the loop. 
+      - When finished, a command is send to the remote tcp server to 
+        cause it to terminate. 
+    """
+
+    name = 'sal_tcpclient'
+
+    def tcp_client_tx_rx(self, ipaddrs, port):
+        "Thread entrypoint for sending and receiving data to/from the remote tcp echo server."
+      
+        s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        s.bind(("", 0))
+        s.connect((ipaddrs, port))
+        
+        tx_data = "1234567890abcdef"
+        for i in xrange(8):
```

Comment:
Agreed. I will define else where. 
Why the value of 8? This is testing data channel connectivity and not intending to stress the channel, which larger values would do.